src/vs/workbench/parts/lib/node/configVariables.ts

```diff
@@ -25,14 +25,16 @@ export class ConfigVariables extends SystemVariables {
 	}
 
 	protected resolveString(value: string): string {
+		const originalValue = value;
 		value = super.resolveString(value);
 
 		let regexp = /\$\{config\.(.*?)\}/g;
 		return value.replace(regexp, (match: string, name: string) => {
 			let config = this.configurationService.getConfiguration();
 			let newValue = new Function('_', 'try {return _.' + name + ';} catch (ex) { return "";}')(config);
```

Comment:
@isidorn , not sure how I could simplify this. Open to suggestions.
The only thing that comes to mind is adding more comments.
The use of new Function syntax is to allow for evaluation of expressions such as ```terminal.integrated.fontFamily``` or even the possibility of array indexes, and the like.